content/en/blog/_posts/XXXX-XX-XX-fine-grained-supplementalgroups-control-beta/index.md

```diff
@@ -0,0 +1,204 @@
+
+```none
+user-defined-in-image:x:1000:
+group-defined-in-image:x:50000:user-defined-in-image
+```
+
+Aha! The container's primary user `1000` belongs to the group `50000` in the last entry.
+
+Thus, the group membership defined in `/etc/group` in the container image for the container's primary user is _implicitly_ merged to the information from the Pod. Please note that this was a design decision the current CRI implementations inherited from Docker, and the community never really reconsidered it until now.
+
+### What's wrong with it?
+
+The _implicitly_ merged group information from `/etc/group` in the container image may cause some concerns particularly in accessing volumes (see [kubernetes/kubernetes#112879](https://issue.k8s.io/112879) for details) because file permission is controlled by uid/gid in Linux. Even worse, the implicit gids from `/etc/group` can not be detected/validated by any policy engines because there is no clue for the implicit group information in the manifest. This can also be a concern for Kubernetes security.
+
+## Fine-grained supplemental groups control in a Pod: `supplementaryGroupsPolicy`
+
+To tackle the above problem, Pod's `.spec.securityContext` now have `supplementalGroupsPolicy` field.
+
+This field provides a way to control how to calculate supplementary groups for the container processes in a Pod. The available policy is below:
+
+* _Merge_: The group membership defined in `/etc/group` for the container's primary user will be merged. If not specified, this policy will be applied (i.e. as-is behavior for backward compatibility).
+
+* _Strict_: it only attaches specified group IDs in `fsGroup`, `supplementalGroups`, or `runAsGroup` fields as the supplementary groups of the container processes. This means no group membership defined in `/etc/group` for the container's primary user will be merged.
+
+Let's see how `Strict` policy works. Below Pod manifest specifies `supplementalGroupsPolicy: Strict`:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: strict-supplementalgroups-policy
+spec:
+  securityContext:
+    runAsUser: 1000
+    runAsGroup: 3000
+    supplementalGroups: [4000]
+    supplementalGroupsPolicy: Strict
+  containers:
+  - name: ctr
+    image: registry.k8s.io/e2e-test-images/agnhost:2.45
+    command: [ "sh", "-c", "sleep 1h" ]
+    securityContext:
+      allowPrivilegeEscalation: false
+```
+
+The result of `id` command in the `ctr` container should be similar to this:
+
+```none
+uid=1000 gid=3000 groups=3000,4000
+```
+
+You can see `Strict` policy can exclude group `50000` from `groups`! 
+
+Thus, ensuring `supplementalGroupsPolicy: Strict` (enforced by some policy mechanism) helps prevent the implicit supplementary groups in a Pod.
+
+{{<note>}}
+Actually, this is not enough because container with sufficient privileges / capability can change its process identity. Please see the following section for details.
```

Comment:
Update the wording to start with the main point, provide some context, and then explain what comes next -- which I find is easier to follow personally. What do you think?

```suggestion
Keep in mind: a container with sufficient privileges can still change its process identity. The `supplementalGroupsPolicy` only affect the initial process identity. See the following section for details.
```